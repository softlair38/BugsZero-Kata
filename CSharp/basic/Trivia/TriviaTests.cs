using Assent;
using System;
using System.IO;
using System.Text;
using Xunit;

namespace Trivia
{
	public class TriviaTests
	{
		[Fact]
		public void RefactoringTests()
		{
			var output = new StringBuilder();
			Console.SetOut(new StringWriter(output));

			Console.WriteLine(false); // test à revoir
			Game aGame = new Game(new Player("Chet"), new Player("Pat"), new Player("Sue"));

			var roll1 = new Roll(1);
			aGame.Roll(roll1);
			aGame.Roll(roll1);
			aGame.Roll(roll1);
			aGame.Roll(roll1);
			aGame.Roll(roll1);
			aGame.Roll(roll1);
			aGame.Roll(roll1);
			aGame.Roll(roll1);
			aGame.Roll(roll1);
			aGame.Roll(roll1);
			aGame.Roll(roll1);
			aGame.Roll(roll1);
			aGame.Roll(roll1);
			aGame.Roll(roll1);

			aGame.WasCorrectlyAnswered(roll1);
			aGame.WrongAnswer();

			var roll2 = new Roll(2);
			aGame.Roll(roll2);

			aGame.Roll(new Roll(6));

			aGame.WrongAnswer();

			aGame.Roll(roll2);

			aGame.Roll(roll2);


			aGame.WrongAnswer();

			aGame.WasCorrectlyAnswered(roll2);
			aGame.Roll(roll1);
			aGame.WasCorrectlyAnswered(roll1);

			var configuration = BuildConfiguration();
			this.Assent(output.ToString(), configuration);
		}

		private static Configuration BuildConfiguration()
		{
			return
				new Configuration()

				// Uncomment this block if an exception 
				// « Could not find a diff program to use »
				// is thrown and if you have VsCode installed.
				// Otherwise, use other DiffProgram with its full path
				// as parameter.
				// See  https://github.com/droyad/Assent/wiki/Reporting
				//                    .UsingReporter(
				//                        new DiffReporter(
				//                            new []
				//                            {
				//                                 For linux
				//                                new VsCodeDiffProgram(new []
				//                                {
				//                                    "/usr/bin/code"
				//                                })

				//                                 For Windows
				//                                new VsCodeDiffProgram(), 
				//                            }))
				;
		}
	}
}
