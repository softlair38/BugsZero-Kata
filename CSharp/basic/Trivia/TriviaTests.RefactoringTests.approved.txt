False
Chet was Added
They are player number 1
Pat was Added
They are player number 2
Sue was Added
They are player number 3
Chet is the current player
They have rolled a 1
Chet's new location is 1
The category is Science
Science Question 0
Chet is the current player
They have rolled a 1
Chet's new location is 2
The category is Sports
Sports Question 0
Chet is the current player
They have rolled a 1
Chet's new location is 3
The category is Rock
Rock Question 0
Chet is the current player
They have rolled a 1
Chet's new location is 4
The category is Pop
Pop Question 0
Chet is the current player
They have rolled a 1
Chet's new location is 5
The category is Science
Science Question 1
Chet is the current player
They have rolled a 1
Chet's new location is 6
The category is Sports
Sports Question 1
Chet is the current player
They have rolled a 1
Chet's new location is 7
The category is Rock
Rock Question 1
Chet is the current player
They have rolled a 1
Chet's new location is 8
The category is Pop
Pop Question 1
Chet is the current player
They have rolled a 1
Chet's new location is 9
The category is Science
Science Question 2
Chet is the current player
They have rolled a 1
Chet's new location is 10
The category is Sports
Sports Question 2
Chet is the current player
They have rolled a 1
Chet's new location is 11
The category is Rock
Rock Question 2
Chet is the current player
They have rolled a 1
Chet's new location is 0
The category is Pop
Pop Question 2
Chet is the current player
They have rolled a 1
Chet's new location is 1
The category is Science
Science Question 3
Chet is the current player
They have rolled a 1
Chet's new location is 2
The category is Sports
Sports Question 3
Answer was correct!!!!
Chet now has 1 Gold Coins.
Question was incorrectly answered
Pat was sent to the penalty box
Sue is the current player
They have rolled a 2
Sue's new location is 2
The category is Sports
Sports Question 4
Sue is the current player
They have rolled a 6
Sue's new location is 8
The category is Pop
Pop Question 3
Question was incorrectly answered
Sue was sent to the penalty box
Chet is the current player
They have rolled a 2
Chet's new location is 4
The category is Pop
Pop Question 4
Chet is the current player
They have rolled a 2
Chet's new location is 6
The category is Sports
Sports Question 5
Question was incorrectly answered
Chet was sent to the penalty box
Sue is the current player
They have rolled a 1
Sue is getting out of the penalty box
Sue's new location is 9
The category is Science
Science Question 4
Answer was correct!!!!
Sue now has 1 Gold Coins.
